tests/side-effects/src/lib.rs

```diff
@@ -1,47 +1,47 @@
 #![allow(dead_code)]
 fn add3(x: u32, y: u32, z: u32) -> u32 {
-    x + y + z
+    x.wrapping_add(y).wrapping_add(z)
```

Comment:
How are these changes related to the PR?
While you're here, add comments what these tests are doing.

Reply:
I change everything from `+` to `wrapping_add` so that I make sure (locally) that things are typechecking in F*, without bothering to add preconditions everywhere so that the addition doesn't overflow!

I want at some point to make our "unit" tests to be typed-checked in the backends at CI, but for now there is no such infrastructure :(

I added comments!